lens/base/contract.rkt

```diff
@@ -13,10 +14,12 @@
            ))
 
 (define (lens/c target/c view/c)
-  (gen-lens/c
-   [lens-view (or/c #f [lens? target/c . -> . view/c])]
-   [lens-set (or/c #f [lens? target/c view/c . -> . target/c])]
-   [focus-lens (or/c #f [lens? target/c . -> . (values view/c [view/c . -> . target/c])])]))
+  (rename-contract
+   (gen-lens/c
+    [lens-view (or/c #f [lens? target/c . -> . view/c])]
+    [lens-set (or/c #f [lens? target/c view/c . -> . target/c])]
+    [focus-lens (or/c #f [lens? target/c . -> . (values view/c [view/c . -> . target/c])])])
+   `(lens/c ,(contract-name target/c) ,(contract-name view/c))))
```

Comment:
This is definitely a good idea. In addition to making the error messages more readable, it also prevents our errors from leaking the information that lenses are defined with generics.